code/modules/paperwork/paperbin.dm

```diff
@@ -73,10 +73,17 @@
 			P = papers[length(papers)]
 			papers.Remove(P)
 		else
-			if(letterhead_type && alert("Choose a style", null,"Letterhead","Blank")=="Letterhead")
-				P = new letterhead_type
-			else
-				P = new /obj/item/paper
+			var/choice = tgui_alert(user, "Choose a style", "Paperbin", list("Letterhead","Blank", "Cancel"))
+			if(isnull(choice) || !Adjacent(user))
+				return
+			switch(choice)
+				if("Letterhead")
+					P = new letterhead_type
+				if("Blank")
+					P = new /obj/item/paper
+				if("Cancel")
+					return
```

Comment:
```suggestion
			if(isnull(P))
				return
```

Reply:
I dont get removal "Cancel" from switch-case

Reply:
but i will add null-check on P right after switch-case just in case

Reply:
this could just be `else` tbh